### PR TITLE
Luke/select bug

### PR DESCRIPTION
**Pull request by lukeschunk**
### First pass at fixing a bug where select component would start from top no matter what
* I don't believe that this is a perfect implementation - I know the callback in `setState` is an odd pattern. Looking for other ideas on how to make this work 
* Also adding an ID and using `getElementById` seems cumbersome but again, I'm still working out a better way. If you use the existing classname `mx-select-options` will not work in this case (2 elements with that class)

### BEFORE: 
* Notice how when an item is selected, then the component is reopened, it starts from the top

select_before

### AFTER:

select_after

### CODE REVIEW

File: src/components/Select.js [hunk 1]
```diff
@@ -66,16 +67,25 @@ const Select = React.createClass({
     if (!isMobile) {
       this.setState({
         isOpen: !this.state.isOpen
+      }, () => {
+        if (document.getElementById('items')) {
+          const items = document.getElementById('items');
+
+          items.scrollTop = this.state.scrollLocation;
+        }
```
Comment (lukeschunk) [inline src/components/Select.js]: I know this is weird to do a callback in `setState` but I haven't figured out a good way to get around it.  If i  do something like this:
```javascript
_handleClick () {
  if (!isMobile) {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }

  this.setScrollPosition();
},

setScrollPosition () {
  if (document.getElementById('items')) {
    const items = document.getElementById('items');

    items.scrollTop = this.state.scrollLocation;
  }
},
```
this.state.scrollLocation` is not set to the proper value when the select component is closed, and thus it won't fix the problem

Comment (tumentumurchudur) [inline src/components/Select.js]: I think what is happening is that when you call `setScrollPosition`, the DOM is already done updating. As soon as the `state` of your component changes, `render` is called to update the component.

Comment (lukeschunk) [inline src/components/Select.js]: thanks for taking a look @tumentumurchudur I'll look into that

Comment (tumentumurchudur) [inline src/components/Select.js]: With that being said, I would move it to the `render` as below:
```javascript
if (this.state.isOpen) {
   const items = document.getElementById('items');

   if (items) {
     items.scrollTop = this.state.scrollLocation;
   }
}
```

**Comment by cerinman** [comment]
@lukeschunk This looks like it never went anywhere.  Can we close this in favor of your select refactor?

**Comment by lukeschunk** [comment]
sure

**Pull request closed without merging**